include/grpc/census.h

```diff
@@ -100,6 +101,150 @@ int census_context_deserialize(const char *buffer, census_context **context);
+    will be the logical or of census_trace_mask_values values. */
+int census_trace_mask(const census_context *context);
+
+/* The concept of "operation" is a fundamental concept for Census. An
+   operation is a logical representation of a action in a RPC-using system. It
+   is most typically used to represent a single RPC, or a significant sub-part
+   thereof (e.g. a single logical "read" RPC to a distributed storage system
+   might do several other actions in parallel, from looking up metadata
+   indices to making requests of other services - each of these could be a
+   sub-operation with the larger RPC operation. Census uses operations for the
+   following:
+
+   CPU accounting: If enabled, census will measure the thread CPU time
+   consumed between operation start and end times.
+
+   Active operations: Census will maintain information on all currently
+   active operations.
+
+   Distributed tracing: Each operation serves as a logical trace span.
+
+   Stats collection: Stats are broken down operation (e.g. latency
+   breakdown for each service/method combination).
+
+   The following functions serve to delineate the start and stop points for
+   each logical operation. */
+/**
+   Start a client rpc operation. This function will create a new context. If
+   the context argument is non-null, then the new context will inherit all
+   its properties, with the following changes:
+   - create a new operation ID for the new context, marking it as a child of
+     the previous operation.
+   - use the new RPC service/method/peer information for tracing and stats
+     collection purposes, rather than those from the original context
+   - if trace_mask is non-zero, update the trace mask entries associated with
+     the original context.
+
+   If the context argument is NULL, then a new root context is created.  This
+   is particularly important for tracing purposes (the trace spans generated
+   will be unassociated with any other trace spans, except those
+   downstream). Whatever it's value, the trace_mask will be used for tracing
+   operations associated with the new context.
+
+   @param context The base context. Can be NULL.
+   @param service RPC service name. On some systems, may include other
+   parts of RPC identification (e.g. host on gRPC systems).
+   @param method RPC method name
+   @param peer RPC peer
+   @param trace_mask An or of census_trace_mask_values values
+   @param start_time If NULL, the time of function call is used as the
+   start time for the operation. If non-NULL, then the time should be in the
+   past, when the operation was deemed to have started. This is used when
+   the other information used as arguments is not yet available.
+
+   @return A new census context.
+ */
+census_context *census_start_client_rpc_op(census_context *context,
+                                           const char *service,
+                                           const char *method, const char *peer,
```

Comment:
Not sure I 100% understand what you're asking, but if it is essentially: can we have the API accept a single argument (where we now have service + method), which are then split later, then yes we can do that. The downside would be that we then have to make other RPC systems conform to whatever encoding we use for that separation (remember, in the long term, we want to have census be usable by other RPC systems). I'd much rather keep the interface here generic.